internal/conditions/valdiate_cluster.go

```diff
@@ -0,0 +1,79 @@
+package conditions
+
+import (
+	aeconditions "github.com/giantswarm/apiextensions/v3/pkg/conditions"
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	capiconditions "sigs.k8s.io/cluster-api/util/conditions"
+
+	"github.com/giantswarm/azure-admission-controller/internal/errors"
+)
+
+func ValidateClusterConditions(oldClusterCR *capi.Cluster, newClusterCR *capi.Cluster) error {
+	var err error
+
+	err = ValidateCreatingCondition(oldClusterCR, newClusterCR)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	err = ValidateUpgradingCondition(oldClusterCR, newClusterCR)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func ValidateCreatingCondition(oldClusterCR *capi.Cluster, newClusterCR *capi.Cluster) error {
+	var err error
+	oldCreatingCondition := capiconditions.Get(oldClusterCR, aeconditions.CreatingCondition)
+	newCreatingCondition := capiconditions.Get(newClusterCR, aeconditions.CreatingCondition)
+
+	// See functions for details about validations.
+	validations := []conditionChangeValidation{
+		validateNewStatusUnknownIsNotAllowed,
+		validateNewStatusMustBeEitherTrueOrFalse,
+		validateRemovingConditionIsNotAllowed,
+	}
+
+	err = validateAll(oldCreatingCondition, newCreatingCondition, validations)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	creationWasCompleted := oldCreatingCondition != nil && oldCreatingCondition.Status == corev1.ConditionFalse
+	nowIsInCreation := newCreatingCondition != nil && newCreatingCondition.Status == corev1.ConditionTrue
```

Comment:
Typo I presume? `s/now/new`

Reply:
no, it was on purpose :smile: the idea was to read it like "if creation was completed and now is in creation" :)

Reply:
:see_no_evil:

Reply:
:smile:

Reply:
basically "if creation was completed and now is in creation again, what heck are you doing, you cannot restart cluster creation after it's done" :smile: